src/Dofs/DofHandler.jl

```diff
@@ -912,7 +912,7 @@ getfieldinterpolation(sdh::SubDofHandler, field_idx::Int) = sdh.field_interpolat
 getfieldinterpolation(sdh::SubDofHandler, field_name::Symbol) = getfieldinterpolation(sdh, find_field(sdh, field_name))
 
 """
-    evaluate_at_grid_nodes(dh::AbstractDofHandler, u::Vector{T}, fieldname::Symbol) where T
+    evaluate_at_grid_nodes(dh::AbstractDofHandler, u::AbstractVector{T}, fieldname::Symbol) where T
 
 Evaluate the approximated solution for field `fieldname` at the node
 coordinates of the grid given the Dof handler `dh` and the solution vector `u`.
```

Comment:
Would it make sense to add some note like this? 
```suggestion
coordinates of the grid given the Dof handler `dh` and the solution vector `u`.
Note that `u` is indexed by the DoF number from `dh`, passing only the solution 
pertinent to `fieldname` will therefore give the wrong result. 
```

Reply:
This is indeed one example. GPU should also work here with some trick, although probably a bit slower.

Reply:
Not sure, it is really difficult to pull the solution vector into subfields. We can just add some assertion in the function which checks if the length is okay and throw an error that the user probably passed the wrong aray.

Reply:
True, `ndofs(dh)==length(u)` should always be fullfilled, so that is a nice check to add! That is probably better than my additional note!

Reply:
Can you add that in a separate PR?